### PR TITLE
provide synced images with multiple "/" in them.

### PR DESCRIPTION
**Pull request by thomasmckay**
https://pulp.plan.io/issues/2368

### CODE REVIEW

**Comment by ipanova** [comment]
ok test

**Comment by seandst** [comment]
The test failures appear to be a result of this change:

https://github.com/thomasmckay/pulp_docker/blob/811108e3b187c5a5473b15e2bbf01151da453a47/plugins/test/unit/plugins/distributors/test_configuration.py#L155 is failing because one of the "valid" repo names is coming back as invalid now. I can't easily tell which one, but I suspect it's `..............`; that is not actually a valid docker repo ID, since periods are not allowed as the first character of a repo id.

The other failure is that a repo ID "noUpperCase" is not raising a validation error for some reason:
https://github.com/thomasmckay/pulp_docker/blob/811108e3b187c5a5473b15e2bbf01151da453a47/plugins/test/unit/plugins/distributors/test_configuration.py#L105

The second test appears to require changing the value of a constant to run, so...I'm not sure what's going on there.

**Comment by thomasmckay** [comment]
I updated the PR after testing the regex locally. If someone could provide specific error failures (since tests are blocked by login/passwd), that would be helpful. Thanks!

**Comment by asmacdo** [comment]
ok test

File: plugins/test/unit/plugins/distributors/test_configuration.py [hunk 1]
```diff
@@ -140,7 +139,7 @@ def test__is_valid_repo_registry_id(self):
             '134567890',
             'alpha-numeric_123',
             'periods.are.cool',
-            '..............',
+            'repo-registry-id'
```
Comment (seandst) [inline plugins/test/unit/plugins/distributors/test_configuration.py]: can we add some values with one and more than one slash in here to ensure we're guarded against regression in multi-slash registry ids when we're directly testing the `_is_valid_repo_registry_id` unit?

File: plugins/test/unit/plugins/distributors/test_configuration.py [hunk 2]
```diff
@@ -149,7 +148,8 @@ def test__is_valid_repo_registry_id(self):
             'upperCase',
             'uppercasE',
             '$ymbols',
-            '$tuff.th@t.m!ght.h@ve.w%!rd.r#g#x.m*anings()'
+            '$tuff.th@t.m!ght.h@ve.w%!rd.r#g#x.m*anings()',
+            '..............'
         ]
```
Comment (seandst) [inline plugins/test/unit/plugins/distributors/test_configuration.py]: We should also have a string of length 256 and a string with length greater than 256, again to guard against regression in the `_is_valid_repo_registry_id` unit.

File: plugins/test/unit/plugins/distributors/test_configuration.py (hunk 1 above)
Comment (seandst) [inline plugins/test/unit/plugins/distributors/test_configuration.py]: Oh, we'll also want to include a registry ID with length 255 that should be valid. I think that's it, though.

**Comment by thomasmckay** [comment]
@seandst - slash and length tests added

**Comment by seandst** [comment]
The f25 test failures are environmental, f24 worked fine.

**Pull request merged**